source-stripe-native/source_stripe_native/resources.py

```diff
@@ -88,221 +143,396 @@ async def all_resources(
+                            all_streams.append(
+                                split_child_object(
+                                    base_stream,
+                                    child_stream,
+                                    http,
+                                    config.start_date,
+                                    platform_account_id,
+                                    connected_account_ids,
+                                )
+                            )
                         case _:
-                            all_streams.append(child_object(base_stream, child_stream, http, config.start_date))
+                            all_streams.append(
+                                child_object(
+                                    base_stream,
+                                    child_stream,
+                                    http,
+                                    config.start_date,
+                                    platform_account_id,
+                                    connected_account_ids,
+                                )
+                            )
 
     # Regional streams are only available in certain countries, so we always have
     # check if they accessible regardless of the API key provided.
     for stream in REGIONAL_STREAMS:
         url = f"{API}/{stream.SEARCH_NAME}"
         if await check_accessibility(http, log, url):
-            resource = base_object(stream, http, config.start_date)
+            resource = base_object(
+                stream,
+                http,
+                config.start_date,
+                platform_account_id,
+                connected_account_ids,
+            )
             all_streams.append(resource)
 
     return all_streams
 
 
+def _create_initial_state(all_account_ids: list[str]) -> ResourceState:
+    cutoff = datetime.now(tz=UTC)
+
+    initial_state = ResourceState(
+        inc={
+            account_id: ResourceState.Incremental(cursor=cutoff)
+            for account_id in all_account_ids
+        },
+        backfill={
+            account_id: ResourceState.Backfill(next_page=None, cutoff=cutoff)
+            for account_id in all_account_ids
+        },
+    )
+
+    return initial_state
+
+
 def base_object(
-    cls, http: HTTPSession, start_date: datetime
+    cls,
+    http: HTTPSession,
+    start_date: datetime,
+    platform_account_id: str,
+    connected_account_ids: list[str],
 ) -> Resource:
     """Base Object handles the default case from source-stripe-native
     It requires a single, parent stream with a valid Event API Type
     """
+    all_account_ids = [platform_account_id, *connected_account_ids]
 
     def open(
         binding: CaptureBinding[ResourceConfig],
         binding_index: int,
         state: ResourceState,
         task: Task,
-        all_bindings
+        all_bindings,
     ):
+        fetch_chnages_fns = {
```

Comment:
Typo in this variable name, and all similar declarations in below it in other resource builder functions.

Reply:
Good catch. Fixed typo. Any recommended extensions/tools for quick checks for spelling errors in code?

Reply:
:shrug: Sometimes fresh eyes are the only thing that catches it.